File: legacy/toexe/menuGenerator.py
import PySimpleGUI as sg
import random
import yaml

def shuffle(obj):
    random.shuffle(obj)
    return str(obj[0])

def menu_generate(text_output):
    settings = {}

    menu = "\n"
    with open("settings.yml", "r+", encoding="utf-8") as stream:
        try:
            settings = yaml.safe_load(stream)  
            for i in range(len(settings)):
                dishes_result=""
                for j in range(settings[i]['count']):
                    dishes_result += (shuffle(settings[i]['values']))
                    if j+1 < settings[i]['count']:
                        dishes_result += ", "
                    if (settings[i]['count'] > 1) and (j!=0) and ((j % 2) != 0) and (j < settings[i]['count']-1):
                        dishes_result += "\n          "
                menu += settings[i]['name'] + ": \n          " + dishes_result+ "\n"
            text_elem = text_output
            text_elem.update("                                                                         МЕНЮ: {}".format(menu))
        except yaml.YAMLError as exc:
            print(exc)
def main():
    layout = [[sg.Button('Сгенерировать меню',enable_events=True, key='-FUNCTION-', font='Helvetica 14',expand_x = True, expand_y = False)],
              [sg.Text("                                                                         МЕНЮ:\n", size=(800, 550), key='-text-', font='Helvetica 14',),sg.Multiline()]]
    window = sg.Window('Генератор меню на неделю', layout, size=(800,550),resizable=True, finalize=True)
    while True:
        event, values = window.read()
        if event in (sg.WIN_CLOSED, 'Exit'):
            break
        elif event == '-FUNCTION-':
            menu_generate(window['-text-'])
    window.close()

if __name__ == "__main__":
    main()

File: legacy/menuGenerator.cs
# подключаем библиотеки
import PySimpleGUI as sg
import random
import yaml


## sdfsfsf sdf sdf sdf 
## что будет внутри окна
## первым описываем кнопку и сразу указываем размер шрифта
#layout = [[sg.Button('Новое число',enable_events=True, key='-FUNCTION-', font='Helvetica 16')],
#        # затем делаем текст
#        [sg.Text('Результат:', size=(25, 1), key='-text-', font='Helvetica 16')]]
## рисуем окно
#window = sg.Window('Генератор случайных чисел', layout, size=(350,100))
## запускаем основной бесконечный цикл
#while True:
#    # получаем события, произошедшие в окне
#    event, values = window.read()
#    # если нажали на крестик
#    if event in (sg.WIN_CLOSED, 'Exit'):
#        # выходим из цикла
#        break
## закрываем окно и освобождаем используемые ресурсы
#window.close()

settings = {}

with open("settings.yaml", "w+") as stream:
    try:
        settings = yaml.safe_load(stream))
    except yaml.YAMLError as exc:
        print(exc)






f = open('settings.yml', 'w+')
try:
   # работа с файлом
    setting


finally:
   f.close()
re

def generate_menu ( settings) :
    for type in setting.







    
dishes_amount = get-content -file .settings.xml

function generate_menu (dishes_amount)
{
dishes_result = @{}
foreach(type in dishes_amount.type)
    rand = new Random();
    for (int i=0;i<dishes_amount.
    rand.Shuffle(type.values);
    dishes_result.add("type","rand")
}